### PR TITLE
Send severe logs and crash reports with Javamail #5771

### LINKED ISSUE #5771: Send severe logs and crash reports with Javamail
As per discussion in #5738

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #5771 

Decommissioned `sendEmailWith(out)Logging` as well because the usages actually coincide with sending normal email or logs/error email. The sole exception is admin email, but I don't really see a drawback of logging it as well.

Probably the only concern here is quota; 100 per day for severe logs compilation and crash reports may or may not be enough. (hopefully enough because it means there are less errors in the system!)

### CODE REVIEW

File: src/main/java/teammates/logic/core/EmailSender.java [hunk 1]
```diff
@@ -22,7 +22,10 @@
     
     private static final Logger log = Utils.getLogger();
     
-    private EmailSenderService service;
+    private final EmailSenderService service;
+    
+    // for sending severe logs compilation and error report
```
Comment (kanghj) [inline src/main/java/teammates/logic/core/EmailSender.java]: We should probably explain here why Javamail is always used for severe logs

**Comment by wkurniawan07** [comment]
@kanghj changes made

**Comment by damithc** [comment]
Can we use the regular service as usual but send a copy using GAE mail? Add a prefix to the subject of the duplicate (e.g. `[Copy]...`

**Comment by wkurniawan07** [comment]
It's workable, but it'll spam the admin's mail box since it effectively doubles the number of error/log report emails sent.

**Comment by damithc** [comment]
It's ok. I'm the one who get spammed. I know I get error messages in triplicate (one from severe logs, one from java mail, one from normal service) which helps me detect if anything goes wrong with one of the three fault reporting mechanisms.

**Pull request merged**